lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -294,6 +294,16 @@ def parse_rates_response(response, origin, destination)
       CPPWSRateResponse.new(true, "", {}, :rates => rates)
     end
 
+    def price_from_node(node, exclude_tax)
+      price = node.at('price-details/due').text
+      return price unless exclude_tax
+      taxes = 0
+      node.at('price-details/taxes').children.each do |node|
+        taxes += Package.cents_from(node.text) if node.elem?
```

Comment:
While it is cleaner we end up iterating over the same list twice (though smaller second time) with my current code we only iterate once. But if we are in favour of not having the block i will implement the cleaner solution.